### PR TITLE
Added info on use of AI in PRs

### PR DESCRIPTION
**Pull request by StandingPadAnimations**
In 2023, the use of AI and LLMs for software development has become more popular. While AI has the potential to be a powerful tool for developers, it can also end up being a hindrance. 

This PR adds general guidelines on using AI for contributions, which aim to be a balance with how AI can be used for development. They are the following:

> - The AI is supplemental to the developer's work, not the other way around.
> - The developer is able to modify the generated code based on the requests given in review.
> - The code remains mostly human written, with the AI being used to generate boilerplate code or to help with repetitive tasks. 
> 
> Avoid the following:
> - Becoming dependent on the AI 
> - Using generated code you don't understand
> - Using AI exclusively without manual work 
> - Etc.

This PR also removes a section on Internal Rewrites (which is now redundant)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,6 +16,20 @@ Thanks for your interest in helping build and extend this addon! MCprep welcomes
...
 
+## Use of AI in Development
+Since 2023, AI (Artificial Intelligence) and LLMs (Large Language Models) have surged in popularity and become more common place in development. AI can be an extremely useful tool for developers, especially when used in conjunction with existing knowledge, but it can also be a hindrance with the regards to the quality of generated code. With these in mind, we allow AI for pull requests with the following conditions:
+- The AI is supplemental to the developer's work, not the other way around.
+- The developer is able to modify the generated code based on the requests given in review.
+- The code remains mostly human written, with the AI being used to generate boilerplate code or to help with repetitive tasks.
```
[the remark this answers is not in the document]
Comment (StandingPadAnimations) [reply CONTRIBUTING.md:23]: I think we could also add the following rule of thumb for developers:
> A good rule of thumb is this: if it's a lazy use of AI given the size of the change, then it's not a good use of AI.

It's not perfect, but it does illustrate what standards we have for using AI

[the remark this answers is not in the document]
Comment (StandingPadAnimations) [reply CONTRIBUTING.md:23]: I think we can also add the following;
> A good rule of thumb is this: if it's a lazy use of AI given the size of the change, then it's not a good use of AI.

It's not perfect, but it's simple to remember and encompasses what we don't want.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -16,27 +16,32 @@ Thanks for your interest in helping build and extend this addon! MCprep welcomes
...
 
-1. Automated tests plus an automated installer makes ensures that any changes that break older versions of blender will be caught automatically.
-1. Abstracting API changes vs directly implementing changes. Instead of swapping "group" for "collection" in the change to blender 2.8, we create if/else statements and wrapper functions that fetch the attribute that exists based on the version of blender. Want more info about this? See [the article here](https://theduckcow.com/2019/update-addons-both-blender-28-and-27-support/).
+A good rule of thumb is this: if it's a lazy use of AI given the size of the change, then it's not a good use of AI.
 
-## Internal Rewrites
-MCprep has a separate branch for internal rewrites based on the dev branch. Sometimes, internal tools are deprecated, and requires features to be changed to reflect those deprecations.
+Avoid the following:
```
Comment (TheDuckCow) [inline CONTRIBUTING.md:29]:
```suggestion
Avoid the following:
- Using AI before having an idea of what change or problem you are solving
```

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheDuckCow) [reply CONTRIBUTING.md:23]: Sure I'm ok with that for now

**Review by TheDuckCow — APPROVED** [review]
Minor suggested addition, but otherwise a good guideline indeed 👍

**Pull request merged**